rpc/wrpc/beacon/src/node.rs

```diff
@@ -0,0 +1,64 @@
+use crate::imports::*;
+use xxhash_rust::xxh3::xxh3_64;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Node {
+    #[serde(skip)]
+    pub id: String,
+    pub name: Option<String>,
+    pub location: Option<String>,
+    pub address: String,
+    pub transport: Transport,
+    pub encoding: WrpcEncoding,
+    pub network: NetworkId,
+    pub enable: Option<bool>,
+    pub bias: Option<f64>,
+    pub version: Option<String>,
+    pub provider: Option<String>,
+    pub link: Option<String>,
+}
+
+impl Eq for Node {}
+
+impl PartialEq for Node {
+    fn eq(&self, other: &Self) -> bool {
+        self.address == other.address
+    }
+}
+
+impl std::fmt::Display for Node {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let title = self.name.clone().unwrap_or(self.address.to_string());
+        write!(f, "{}", title)
+    }
+}
+
+impl Node {
+    pub fn params(&self) -> Params {
+        Params::new(self.encoding, self.network)
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NodeConfig {
+    node: Vec<Node>,
```

Comment:
field name in plural form?

Reply:
Hm... This is TOML deser, as such fields need to be `[[node]]`... OTOH I guess this makes sense and I'll just add `#[serde(rename = "node")]`...